File: Server/Models/Slime.cs
using System.ComponentModel.DataAnnotations;

namespace Server.Models
{
    public class Slime
    {
        public Slime() { }

        public Slime(int id, string name, int size, string color, bool isonmarket, int price, int? ownerid, string? ownername, SlimeStats slimestats)
        {
            Id = id;
            Name = name;
            Size = size;
            Color = color;
            IsOnMarket = isonmarket;
            Price = price;
            OwnerId = ownerid;
            OwnerName = ownername;
            SlimeStats = slimestats;
        }

        [Key] public int Id { get; set; }
        public string Name { get; set; } = "";
        public int Size { get; set; }
        public string Color { get; set; } = "";
        public bool IsOnMarket { get; set; } 
        public int Price { get; set; }
        public int? OwnerId { get; set; } 
        public string? OwnerName { get; set; }
        public SlimeStats SlimeStats { get; set; } = new();
        public string? Svg { get; set; }
    }
}

File: Client/src/types/Slime.ts
import { SlimeStats } from "./SlimeStats";

export type Slime = {
    id: number;
    name: string;
    svg: string;
    size: number;
    color: string;
    isOnMarket: boolean;
    price: number;
    ownerId: number;
    ownerName: string;
    slimeStats: SlimeStats | undefined;
};

export type MarketSlime = {
    id: number;
    name: string;
    svg: string;
    price: number;
    ownerId: number;
    ownerName: string;
};


export type SlimeEditable = {
    id: number;
    name: string;
    isOnMarket: boolean;
    ownerId: number;
    ownerName: string;
}
